### PR TITLE
Add initial support for Floodlight

### PR DESCRIPTION
**Pull request by briis**
Still missing services to:
* Turn on/off the light
* Set brightness
* Set motion mode

### CODE REVIEW

File: pyunifiprotect/unifi_data.py [hunk 1]
```diff
@@ -250,6 +318,30 @@ def event_from_ws_frames(state_machine, minimum_score, action_json, data_json):
...
+
+def light_update_from_ws_frames(state_machine, host, action_json, data_json):
+    """Convert a websocket frame to internal format."""
+
+    if action_json["modelKey"] != "light":
+        raise ValueError("Model key must be light")
+
+    light_id = action_json["id"]
+
+    if not state_machine.has_camera(light_id):
```
Comment (bdraco) [inline pyunifiprotect/unifi_data.py:331]:
```suggestion
    if not state_machine.has_device(light_id):
```

Maybe change `has_camera` to `has_device` (and all other instances)?

File: pyunifiprotect/unifi_data.py [hunk 2]
```diff
@@ -233,7 +299,8 @@ def event_from_ws_frames(state_machine, minimum_score, action_json, data_json):
 
     if action == "add":
         camera_id = data_json.get("camera")
```
Comment (bdraco) [inline pyunifiprotect/unifi_data.py:301]:
```suggestion
        device_id = data_json.get("camera", data_json.get("light"))
```

File: pyunifiprotect/unifi_data.py [hunk 3]
```diff
@@ -233,7 +299,8 @@ def event_from_ws_frames(state_machine, minimum_score, action_json, data_json):
...
-        if camera_id is None:
+        light_id = data_json.get("light")
+        if camera_id and light_id is None:
```
Comment (bdraco) [inline pyunifiprotect/unifi_data.py:303]:
```suggestion
        if device_id is None:
```

File: pyunifiprotect/unifi_data.py [hunk 4]
```diff
@@ -242,6 +309,7 @@ def event_from_ws_frames(state_machine, minimum_score, action_json, data_json):
         if not event:
             return None, None
         camera_id = event.get("camera")
+        light_id = event.get("light")
```
Comment (bdraco) [inline pyunifiprotect/unifi_data.py:312]:
```suggestion
        device_id = event.get("camera", event.get("light"))
```

File: pyunifiprotect/unifi_data.py [hunk 5]
```diff
@@ -250,6 +318,30 @@ def event_from_ws_frames(state_machine, minimum_score, action_json, data_json):
 
     return camera_id, processed_event
```
Comment (bdraco) [inline pyunifiprotect/unifi_data.py:319]:
```suggestion
    return device_id, processed_event
```

File: pyunifiprotect/unifi_protect_server.py [hunk 6]
```diff
@@ -979,6 +1018,26 @@ def _process_camera_ws_message(self, action_json, data_json):
...
+            )
+            if processed_event is not None:
+                _LOGGER.debug("Processed light event: %s", processed_event)
+                processed_light.update(processed_event)
+
+        self.fire_event(light_id, processed_light)
+
     def _process_event_ws_message(self, action_json, data_json):
         """Process a decoded event websocket message."""
         camera_id, processed_event = event_from_ws_frames(
```
Comment (bdraco) [inline pyunifiprotect/unifi_protect_server.py:1043]:
```suggestion
        device_id, processed_event = event_from_ws_frames(
```

Comment (bdraco) [reply pyunifiprotect/unifi_protect_server.py:1043]: ... and update the ones below

File: pyunifiprotect/unifi_protect_server.py [hunk 7]
```diff
@@ -957,6 +992,10 @@ def _process_ws_message(self, msg):
             self._process_camera_ws_message(action_json, data_json)
             return
 
+        if model_key == "light":
+            # TODO:
```
Comment (bdraco) [inline pyunifiprotect/unifi_protect_server.py:996]:
```suggestion
```

File: pyunifiprotect/unifi_data.py [hunk 8]
```diff
@@ -250,6 +318,30 @@ def event_from_ws_frames(state_machine, minimum_score, action_json, data_json):
...
+
+    light_id = action_json["id"]
+
+    if not state_machine.has_camera(light_id):
+        _LOGGER.debug("Skipping non-adopted light: %s", data_json)
+        return None, None
+
+    light = state_machine.update(light_id, data_json)
+
+    if data_json.keys().isdisjoint(CAMERA_KEYS):
```
Comment (bdraco) [inline pyunifiprotect/unifi_data.py:337]:
```suggestion
    if data_json.keys().isdisjoint(LIGHT_KEYS):
```

and add `LIGHT_KEYS` to list the keys we care about.

**Comment by bdraco** [comment]
I added a few suggestions.  I'm just about done for the night though so I'll look back in the morning 👍

**Comment by briis** [comment]
Thank you. I will go through them.
Sleep well.

**Pull request merged**